core/src/main/java/org/apache/accumulo/core/file/rfile/RFile.java

```diff
@@ -1662,23 +1625,8 @@ public Value getTopValue() {
     }
 
     @Override
-    public Text getFirstRow() throws IOException {
```

Comment:
This is where one of the two bugs was.  This method and getLastRow could compute a last row that was before the start row when the fence did not overlap the first and last row in the file.  Now the new code should return an empty FileRange for this case.